File: LeetCode/Medium/0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        sum = 0
        arr = []
        for i in nums:
            sum += i
            arr.append(sum)

        st = 0
        en = st 
        ret = sys.maxsize
        while st < len(nums) and en < len(nums):
            if arr[en] - arr[st] + nums[st] >= target:
                ret = min(ret, en - st + 1)
            if arr[en] - arr[st] + nums[st] > target:
                st += 1
            elif arr[en] - arr[st] + nums[st] < target:
                en += 1
            else:
                st += 1
                en += 1

        if ret == sys.maxsize:
            return 0
        else:
            return ret

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int n = nums.size();
        int ret = INT_MAX;
        int sum = 0;
        int l = 0;

        for (int r = 0; r < n; r++){
            sum += nums[r];
            while (sum >= target){
                ret = min(ret, r - l + 1);
                sum -= nums[l];
                l++;
            }
        }
        return ret == INT_MAX ? 0 : ret;
    }
};